tools/data_integrity/score_instrument.php

```diff
@@ -0,0 +1,241 @@
+        && !is_file(__DIR__."/../../project/instruments/$test_name.linst")
+    ) {
+        printError("Included file does not exist (".__DIR__."/../../project/instruments/NDB_BVL_Instrument_$test_name.class.inc)");
+        return false;
+    }
+}
+
+// check that the $test_name is a valid instrument
+if($test_name== 'all') {
+    $query = "SELECT DISTINCT test_name FROM test_battery WHERE Active='Y'";
+} else {
+    $query = "SELECT DISTINCT test_name FROM test_battery WHERE Active='Y' AND Test_name = :tnm";
+}
+$testNames = $DB->pselect($query, array('tnm' => $test_name));
+
+// if nothing is returned than the instrument DNE
+if (!is_array($testNames) || count($testNames)==0) {
+    printError("Invalid Instrument ($test_name)!");
+    return false;
+}
+
+// get the list of CommentIDs for valid timepoints
+foreach($testNames as $test) {
+    $instrumentName = $test['test_name'];
+    printOut("Running scoring for $instrumentName");
+    logMessage("------------------------------");
+
+    $query = "
+        SELECT s.CandID, 
+            s.Visit_label, 
+            s.ID as SessionID, 
+            f.CommentID, 
+            c.PSCID
+        FROM candidate c 
+            JOIN session s ON c.CandID=s.CandID
+            JOIN flag f ON s.ID=f.SessionID
+        WHERE s.Active = 'Y' 
+            AND c.Active='Y' 
+            AND f.Test_name = :tnm 
+            AND f.Administration <> 'None' 
+            AND f.Administration IS NOT NULL
+            ";
+
+    $params = array('tnm' => $instrumentName);
+
+    if ($action=='one') {
+        $query .= " AND s.ID = :sid AND s.CandID= :cid";
+        $params = array_merge(
+            $params,
+            array(
+             'sid' => $sessionID,
+             'cid' => $candID,
+            )
+        );
+    }
+    $instrumentMetaData = $DB->pselect($query, $params);
+
+    // return error if no candidates/timepoint matched the args
+    if (!is_array($instrumentMetaData) || empty($instrumentMetaData)) {
+        // given that the tool might be run with all, $candID and $sessionID might not be defined
+        if (!empty($candID) && !empty($sessionID)) {
+            printOut(
+                "No records match the criteria returned for ".
+                "candidate ($candID), timepoint ($sessionID), instrument ($instrumentName)!"
+            );
+        } else {
+            printOut("No records for instrument $instrumentName!");
+        }
+        continue;
+    }
+
+    printOut("Start");
+
+    // loop the list and derive scores for each record
+    foreach ($instrumentMetaData as $record) {
+        // make an instance of the instrument's object
+        $instrument      = NDB_BVL_Instrument::factory($instrumentName, $record['CommentID'], '');
+        $instrumentTable = $instrument->table;
+
+        // check if the instrument has a scoring method
+        if (!method_exists($instrument, "score")) {
```

Comment:
Given that the function exists in the base class, how is it possible for it to have gotten removed?

Reply:
my guess is that this is from back when the function was not in the base class ??? possible ?

I'll remove it anyways

Reply:
That would make sense, I guess

Reply:
done